cmd/thanos/tools_bucket.go

```diff
@@ -710,3 +711,44 @@ func compare(s1, s2 string) bool {
 	}
 	return s1Time.Before(s2Time)
 }
+
+func registerBucketMarkBlock(app extkingpin.AppClause, objStoreConfig *extflag.PathOrContent) {
+	cmd := app.Command(component.Mark.String(), "Mark block for deletion or no-compact in a safe way. NOTE: If the compactor is currently running compacting same block, this operation would be potentially a noop.")
+	blockIDs := cmd.Flag("id", "ID (ULID) of the blocks to be marked for deletion (repeated flag)").Required().Strings()
+	// TODO(bwplotka): Add no-compact option once https://github.com/thanos-io/thanos/pull/3409 is merged.
+	// marker := cmd.Flag("marker", "").Required().Enum(metadata.DeletionMarkFilename)
+	cmd.Setup(func(g *run.Group, logger log.Logger, reg *prometheus.Registry, _ opentracing.Tracer, _ <-chan struct{}, _ bool) error {
+		confContentYaml, err := objStoreConfig.Content()
+		if err != nil {
+			return err
+		}
+
+		bkt, err := client.NewBucket(logger, confContentYaml, reg, component.Cleanup.String())
+		if err != nil {
+			return err
+		}
+
+		var ids []ulid.ULID
+		for _, id := range *blockIDs {
+			u, err := ulid.Parse(id)
+			if err != nil {
+				return errors.Errorf("id is not a valid block ULID, got: %v", id)
+			}
+			ids = append(ids, u)
+		}
+
+		ctx, cancel := context.WithTimeout(context.Background(), 2*time.Minute)
```

Comment:
Maybe it's only me but I'd simply use `context.Background()` + https://golang.org/pkg/context/#WithCancel. Maybe someone has such a bad connection that it might take more than two minutes, I don't know :shrug:

Reply:
:+1: